code/modules/mob/living/carbon/xenomorph/castes/praetorian/abilities_praetorian.dm

```diff
@@ -467,3 +467,551 @@ GLOBAL_LIST_INIT(acid_spray_hit, typecacheof(list(/obj/structure/barricade, /obj
+
+	xeno_owner.spin(4, 1)
+	xeno_owner.emote("tail")
+	playsound(xeno_owner, 'sound/weapons/alien_claw_block.ogg', 50, 1)
+	succeed_activate()
+	add_cooldown()
+
+/// Does brute/oxygen damage to unconscious humans if they move.
+/datum/action/ability/activable/xeno/tail_lash/proc/on_movement_while_thrown(datum/source)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/human/human_mob = source
+	if(human_mob.stat != UNCONSCIOUS)
+		UnregisterSignal(human_mob, COMSIG_MOVABLE_MOVED)
+		UnregisterSignal(human_mob, COMSIG_MOVABLE_POST_THROW)
+		return
+	if(human_mob.adjustOxyLoss(HUMAN_CRITDRAG_OXYLOSS))
+		return
+	INVOKE_ASYNC(human_mob, TYPE_PROC_REF(/mob/living/carbon/human, adjustBruteLoss), HUMAN_CRITDRAG_OXYLOSS)
+
+/// Removes signals for COMSIG_MOVABLE_MOVED and COMSIG_MOVABLE_POST_THROW for thrown humans.
+/datum/action/ability/activable/xeno/tail_lash/proc/on_throw_end(datum/source)
+	SIGNAL_HANDLER
+	UnregisterSignal(source, COMSIG_MOVABLE_MOVED)
+	UnregisterSignal(source, COMSIG_MOVABLE_POST_THROW)
+
+// ***************************************
+// *********** Advance
+// ***************************************
+/datum/action/ability/activable/xeno/advance
+	name = "Advance"
+	action_icon_state = "advance"
+	action_icon = 'icons/Xeno/actions/praetorian.dmi'
+	desc = "Launch yourself with tremendous speed toward a location. If you hit a marine, they are launched incredibly far."
+	ability_cost = 50
+	cooldown_duration = 12 SECONDS
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_ADVANCE,
+	)
+
+/datum/action/ability/activable/xeno/advance/use_ability(atom/target)
+	xeno_owner.face_atom(target)
+	if(!do_after(xeno_owner, 0.8 SECONDS, IGNORE_HELD_ITEM, xeno_owner, BUSY_ICON_DANGER))
+		return
+	xeno_owner.face_atom(target)
+
+	RegisterSignal(xeno_owner, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
+	RegisterSignal(xeno_owner, COMSIG_XENO_OBJ_THROW_HIT, PROC_REF(obj_hit))
+	RegisterSignal(xeno_owner, COMSIG_XENOMORPH_LEAP_BUMP, PROC_REF(mob_hit))
+	RegisterSignal(xeno_owner, COMSIG_MOVABLE_POST_THROW, PROC_REF(charge_complete))
+	xeno_owner.xeno_flags |= XENO_LEAPING
+
+	xeno_owner.throw_at(target, 5, 5, xeno_owner)
+	xeno_owner.emote("roar")
+	succeed_activate()
+	add_cooldown()
+
+
+/// Shake the turf under for cool points.
+/datum/action/ability/activable/xeno/advance/proc/on_move(datum/source)
+	SIGNAL_HANDLER
+	var/turf/current_turf = get_turf(source)
+	current_turf.Shake(duration = 0.2 SECONDS)
+
+/// Ends the charge when hitting an object.
+/datum/action/ability/activable/xeno/advance/proc/obj_hit(datum/source, obj/obj_hit, speed)
+	SIGNAL_HANDLER
+	obj_hit.hitby(xeno_owner, speed)
+	charge_complete()
+
+/// Ends the charge when hitting a human. Knocks them back pretty far.
+/datum/action/ability/activable/xeno/advance/proc/mob_hit(datum/source, mob/living/living_hit)
+	SIGNAL_HANDLER
+	if(!ishuman(living_hit))
+		return
+
+	var/mob/living/carbon/human/human_hit = living_hit
```

Comment:
why are you typecasting here? The procs you call are all at the living (or AM) level.